source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -72,23 +58,22 @@ void Filter::initiateCall(const Http::HeaderMap& headers) {
       callbacks_, headers, std::move(context_extensions), check_request_,
       config_->maxRequestBytes());
 
+  ENVOY_STREAM_LOG(trace, "ext_authz filter calling authorization server", *callbacks_);
   state_ = State::Calling;
-  // Don't let the filter chain continue as we are going to invoke check call.
-  filter_return_ = FilterReturn::StopDecoding;
+  filter_return_ = FilterReturn::StopDecoding; // Don't let the filter chain continue as we are
+                                               // going to invoke check call.
   initiating_call_ = true;
-  ENVOY_STREAM_LOG(trace, "ext_authz filter calling authorization server", *callbacks_);
   client_->check(*this, check_request_, callbacks_->activeSpan());
   initiating_call_ = false;
 }
 
 Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool end_stream) {
   request_headers_ = &headers;
-  const auto* method = headers.Method();
   buffer_data_ = config_->withRequestBody() &&
-                 !(end_stream || (method && isHeaderOnlyMethod(method->value().getStringView())));
-
+                 !(end_stream || Http::Utility::isWebSocketUpgradeRequest(headers) ||
```

Comment:
Do filters get upgrade requests? I roughly remember this is handled by HCM now? cc @alyssawilk

Reply:
Apparently, they do but would be nice to hear from @alyssawilk. If this is ok, the same logic ^^ should be applied to the buffer filter.

Reply:
Ah nvm I was in the old impression before we have new style websocket.